tests/data_access/builders/test_todo_entry_builder.py

```diff
@@ -0,0 +1,26 @@
+from datetime import datetime
+
+import pytest
+
+from src.data_access.model_builders.todo_builder import TodoBuilder
+from src.middleware.exceptions.todos import BadServerConfiguration
+
+import src.data_access.model_builders.todo_builder as todo_builder
+
+class TestTodoEntryBuilder:
+
+    def test_set_date_time_parses_string_with_configured_format(self, mocker):
+        mocker.patch.object(todo_builder,"get_config", return_value={"DATETIME_FORMAT": "something"})
+        expected_datetime_parsed = datetime.today()
+        mocked_datetime = mocker.patch.object(todo_builder,"dt")
+        mocked_datetime.strptime.return_value = expected_datetime_parsed
+        builder = TodoBuilder(1, "title").set_date_time("some date time")
+
+        assert builder.build().date_time == expected_datetime_parsed
```

Comment:
what if there is a bug in build function? u should assert the variable _date_time that u set